src/App.js

```diff
@@ -26,13 +26,22 @@ class App extends React.Component {
 
     // TODO: add the new Todo to the list in state
     // make sure not to mutate the existing array!
+    const newTodos = [...this.state.todos, newTodo]
+    this.setState(({
+      todos: newTodos
+    }))
+    console.log("new", this.state)
   }
 
   deleteTodo = id => {
     console.log("deleteTodo", id)
-
+    
     // TODO: remove ONE todo from state using the id
     // make sure not to mutate the existing array!
+    const updatedAfterDelete = this.state.todos.filter(todo => todo.id !== id)
```

Comment:
Here, this works because filter returns a new array - so also, not mutating state.